### PR TITLE
Set up ssl on the nodes

### PR DESCRIPTION
**Pull request by sac**

### CODE REVIEW

File: roles/gluster_hci/tasks/ssl-setup.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  register: result
+  delegate_to: 127.0.0.1
+  run_once: True
+
+- name: Get the name of the directory created
+  set_fact:
+     ssl_base_dir: "{{ result['path'] }}"
+  run_once: True
+
+- name: Generate private key on the servers
```
Comment (sabose) [inline roles/gluster_hci/tasks/ssl-setup.yml]: does this run for all servers in inventory?

File: roles/gluster_hci/tasks/ssl-setup.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     state: restarted
+
+- name: Set volume options for SSL
+  gluster_volume:
+     state: present
+     volume: "{{ item }}"
+     options:
+         { client.ssl: 'on',
+           server.ssl: 'on',
+           auth.ssl-allow: "{{ ssl_allow_list }}" }
```
Comment (sabose) [inline roles/gluster_hci/tasks/ssl-setup.yml]: if ssl_allow_list is empty are the inventory_hosts used as clients?

File: roles/gluster_hci/README.md [hunk 3]
```diff
@@ -26,6 +26,14 @@ Role Variables
 |gluster_features_512B_check|true/false|true|If set to false, logical block size is not checked. By default checks if logical block size for a disk is set to 512 bytes.|
 |gluster_features_slice_setup|true/false|true|If set to false, slice setup will not be done. This features creates a CPU slice limiting the glusterd process's CPU consumption|
 |gluster_features_enable_ipv6|true/false|false|If set to true, role will configure glusterd to use ipv6 address|
+|gluster_features_enable_ssl|true/false|false|If set to true, role will configure ssl on the provided volumes|
+|gluster_features_ssl_clients||None|A comman separated list of ssl clients to be configured. This vairable is optional.|
```
[the remark this answers is not in the document]
Comment (sac) [reply roles/gluster_hci/README.md]: Fixed.

File: roles/gluster_hci/README.md [hunk 4]
```diff
@@ -26,6 +26,14 @@ Role Variables
...
+|gluster_features_priv_key||/etc/ssl/glusterfs.key|If set, the value is used to store the generated key.|
```
[the remark this answers is not in the document]
Comment (sac) [reply roles/gluster_hci/README.md]: No, it has to be a path on the system. User/cockpit-program/some-other-program should have placed the key on the system. And also set the variable gluster_features_ssl_self_signed: false.

File: roles/gluster_hci/tasks/ssl-setup.yml (hunk 1 above)
Comment (sac) [reply roles/gluster_hci/tasks/ssl-setup.yml]: Yes.

File: roles/gluster_hci/tasks/ssl-setup.yml (hunk 2 above)
Comment (sac) [reply roles/gluster_hci/tasks/ssl-setup.yml]: ssl_allow_list is internal variable. Set by us:

```
- set_fact:
      ssl_allow_list: "{{ (groups['all'] + gluster_features_ssl_clients |
                           default([])) | join(',')}}"
```

Comment (sac) [reply roles/gluster_hci/tasks/ssl-setup.yml]: For the user settable variables, they will be prepended by gluster_features ...

File: roles/gluster_hci/tasks/hosted_engine_pre_check.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+---
```
[the remark this answers is not in the document]
Comment (sac) [reply roles/gluster_hci/tasks/hosted_engine_pre_check.yml]: No, it is not. I will rebase and resubmit.

File: roles/gluster_hci/README.md [hunk 6]
```diff
@@ -26,6 +26,14 @@ Role Variables
...
+|gluster_features_ssl_self_signed|true/false|true|If set to false, a private key has to be provided. And the role will not generate a key.|
+|gluster_features_cert_file||/etc/ssl/glusterfs.pem|Path to file where the generated signed certificate has to be placed|
```
[the remark this answers is not in the document]
Comment (sac) [reply roles/gluster_hci/README.md]: The gluster_features_cert_file has to point to third party certificates (if any), then the third party certificate will be used. Else, /etc/ssl/glusterfs.pem will be used (which is self-signed).

File: roles/gluster_hci/tasks/hosted_engine_pre_check.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (sac) [reply roles/gluster_hci/tasks/hosted_engine_pre_check.yml]: It was part of earlier commit. No longer part of this PR.

File: roles/gluster_hci/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (sabose) [reply roles/gluster_hci/README.md]: the documentation is not clear on this part. can you update that?

[the remark this answers is not in the document]
Comment (sac) [reply roles/gluster_hci/README.md]: > the documentation is not clear on this part. can you update that?

Done.

**Pull request merged**